src/App.tsx

```diff
@@ -1,61 +1,156 @@
+import { ChangeEvent, useState } from 'react';
 import './App.scss';
+import { TodoList } from './components/TodoList';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+
+function getUser(userId: number) {
+  const foundUser = usersFromServer.find(user => user.id === userId);
+
+  return foundUser || null;
+}
+
+export const todos = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUser(todo.userId),
+}));
 
 export const App = () => {
+  const [title, setTitle] = useState('');
+  const [user, setUser] = useState('');
+  const [updatedTodos, setUpdatedTodos] = useState(todos);
+  const [noUser, setNoUser] = useState(false);
+  const [noTitle, setNoTitle] = useState(false);
+
+  const maxTodoId = Math.max(
+    ...updatedTodos.map(updatedTodo => updatedTodo.id),
+  );
+
+  function spellingChecker(text:string) {
+    const result = text.match(/[a-zA-Zа-яА-Я\d\s]/g);
+
+    return result ? result.join('') : '';
+  }
+
+  function submitHandler() {
+    if (!user) {
+      setNoUser(true);
+    }
+
+    if (!title.trim()) {
+      setNoTitle(true);
+    }
```

Comment:
Another option but it's up to you
```suggestion
    setNoUser(!user);
    setNoTitle(!title.trim());
```